subsys/bluetooth/mesh/time_srv.c

```diff
@@ -486,7 +520,10 @@ int bt_mesh_time_srv_time_status_send(struct bt_mesh_time_srv *srv,
 		return -EOPNOTSUPP;
 	}
 
-	return send_time_status(srv->model, ctx, k_uptime_get());
+	srv->model->pub->ttl = 0;
+	srv->data.timestamp = uptime;
+
```

Comment:
I would rather move these changes to "send_time_status()". Because "bt_mesh_time_srv_time_status_send()" is meant for sending unsoliciated Time_Status messages and this function can get blocked if there is no correct role.